azurerm/resource_arm_cosmosdb_mongo_collection.go

```diff
@@ -243,11 +258,14 @@ func resourceArmCosmosDbMongoCollectionRead(d *schema.ResourceData, meta interfa
 
 	throughputResp, err := client.GetMongoDBCollectionThroughput(ctx, id.ResourceGroup, id.Account, id.Database, id.Collection)
 	if err != nil {
-		return fmt.Errorf("Error reading Throughput on Cosmos Mongo Collection %s (Account %s, Database %s): %+v", id.Collection, id.Account, id.Database, err)
-	}
-
-	if throughput := throughputResp.Throughput; throughput != nil {
-		d.Set("throughput", int(*throughput))
+		if !utils.ResponseWasNotFound(throughputResp.Response) {
+			_ = d.Set("throughput", nil)
```

Comment:
I don't think we need to set this?

Reply:
I now moved it to the case where the throughput request failed with a 404 indicating that the resource does not have throughput configured. I came across a case while testing, where I created a resource without throuhput, then tried to set it (which is not allowed) resulting in an error but the throughput was still set to the not applied value in the terraform state.